frontend/javascripts/oxalis/view/viewport_status_indicator.js

```diff
@@ -43,9 +49,21 @@ export default function ViewportStatusIndicator() {
     <Tooltip
       title={
         <div>
-          The layer{pluralS} {unrenderableLayerNames.map(name => `"${name}"`).join(", ")} {dontVerb}{" "}
-          exist in the current resolution {currentResolution.join("-")}. Adjust the zoom level to
-          change the active resolution.{renderMissingDataBlackHint}
+          {unrenderableLayerNamesWithSmallerZoomstep.length > 0
+            ? `Zoom in to see layer${pluralSForSmallerZoomstep} ${unrenderableLayerNamesWithSmallerZoomstep.join(
+                ", ",
+              )}${
+                unrenderableLayerNamesWithHigherZoomstep.length > 0 || renderMissingDataBlackHint
+                  ? " "
+                  : ""
+              }`
+            : null}
+          {unrenderableLayerNamesWithHigherZoomstep.length > 0
+            ? `Zoom out to see layer${pluralSForHigherZoomstep} ${unrenderableLayerNamesWithHigherZoomstep.join(
+                ", ",
+              )}${renderMissingDataBlackHint ? " " : ""}`
+            : null}
+          {renderMissingDataBlackHint}
```

Comment:
Do you have any suggestions on how to improve this @philippotto?

Reply:
How about something like this (not tested obviously):

```javascript
function getOptionalZoomMessage(directionString, unrenderableLayersWithInfo) {
  const unrenderableLayerNamesWithOtherZoomstep = unrenderableLayersWithInfo
    .filter(([, { higher, smaller }]) => directionString === "in" ? smaller : higher)
    .map(([layerName]) => layerName);
  if (unrenderableLayerNamesWithOtherZoomstep.length === 0) return null;
  
  const pluralS = unrenderableLayerNamesWithOtherZoomstep.length > 1 ? "s" : "";
  return `Zoom {directionString} to see layer${pluralS} ${unrenderableLayerNamesWithOtherZoomstep.join(
    ", ",
  )}`
}


const zoomInMessage = getOptionalZoomMessage("in", unrenderableLayersWithInfo);
const zoomOutMessage = getOptionalZoomMessage("out", unrenderableLayersWithInfo);
const missingDataHint = ...;
const sentencesString = _.flatten([zoomInMessage, zoomOutMessage, missingDataHint]).join(" ");
...
return <div>{sentencesString}</div>;
```

this would also get rid of the naming problem, since the in/out distinction is not necessary anymore (due to the helper function)

Reply:
Very nice suggestion :smile: Thanks a lot :t-rex: